_overviews/cheatsheets/index.md

```diff
@@ -464,25 +460,26 @@ println(upper.getOrElse(""))</code></pre></td>
   case Some(x) =&gt; f(x)
   case None    =&gt; None
 }</code></pre></td>
-      <td>Same as map but function must return an optional value</td>
+      <td>Same as map but function must return an optional value.</td>
     </tr>
     <tr>
       <td><pre class="highlight"><code>optionOfOption.flatten</code></pre>
       <em><strong>same as</strong></em>
       <pre class="highlight"><code>optionOfOption match {
-  case Some(Some(x)) =&gt; Some(x)
-  case _             =&gt; None
+  case Some(x: Option[_]) =&gt; x
+  case None               =&gt; None
 }</code></pre></td>
-      <td>Extract nested option</td>
+      <td>Extract nested option.</td>
     </tr>
     <tr>
       <td><pre class="highlight"><code>option.foreach(f(_))</code></pre>
       <em><strong>same as</strong></em>
       <pre class="highlight"><code>option match {
   case Some(x) =&gt; f(x)
   case None    =&gt; ()
-}</code></pre></td>
-      <td>Apply a procedure on optional value</td>
+}
+()</code></pre></td>
```

Comment:
Although accurate, putting `()` unit literal after the pattern match seems clumsy.  An annotation to the first pattern match case could be added:

```scala
  case Some(x) => f(x): Unit
```

However, I'm not sure if such precision is necessary on a cheatsheet.

Reply:
I also don't like the added line in the code, but I think it is probably better to leave a note in description that return value is `Unit`, so one can't chain calls like `o.foreach(…).foreach(…)`

Reply:
Sure, but fortunately the compiler will complain if someone tries to chain the calls.

Reply:
Good point. So you think I should remove the note from description?

Reply:
No, I don't think it's necessary.  The term "procedure" is meant to connote that it returns `Unit`.

Reply:
True, removed the note